### PR TITLE
Add tracepoints for query pipeline introspection

### PR DESCRIPTION
**Pull request by lava**
### :memo: Checklist

- [ ] All user-facing changes have changelog entries.
- [x] The changes are reflected on [docs.tenzir.com/vast](https://docs.tenzir.com/vast), if necessary.
- [x] The PR description contains instructions for the reviewer, if necessary.

### :dart: Review Instructions

Review commit-by-commit. For testing:

```
  sudo ./scripts/query-tracing.bcc build/lib/libvast.so
  ./bin/vast -N export json
```

### CODE REVIEW

File: scripts/query-tracing.bcc [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+        ("index_resume", index_resume_event),
+    ]
+
+
+class perf_output(ctypes.Structure):
+    _fields_ = [("query_id", c_ulonglong), ("type_", c_byte), ("event", events)]
+
+
+# Header (use R "long" format for easier subsequent plotting)
+print("variable\tvalue\tquery_id")
```
Comment (tobim) [inline scripts/query-tracing.bcc:224]: Why didn't you also include the R script into this PR?

File: scripts/query-tracing.bcc [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+
+from __future__ import print_function
+from bcc import BPF, USDT
+import ctypes
+from ctypes import c_int, c_ulonglong, c_byte, c_ushort
+from sys import argv
+
+
+def usage():
+    print("USAGE: %s /path/to/libvast.so" % argv[0])
```
Comment (tobim) [inline scripts/query-tracing.bcc:14]: Should probably be piped to a `query_timings.tsv` file?

File: scripts/query-tracing.bcc [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+u.enable_probe("query_partition_error", "query_partition_error_handler")
+u.enable_probe("query_done", "query_done_handler")
+
+# load BPF program
+b = BPF(
+    text="""
+#include <uapi/linux/ptrace.h>
+
+enum event_types {
+    EVENT_QUERY_STATE = 0,
```
Comment (tobim) [inline scripts/query-tracing.bcc:40]: Should this be named `EVENT_QUERY_DONE`?

**Review by tobim — CHANGES_REQUESTED** [review]
Some miner change request and questions.

File: scripts/query-tracing.bcc (hunk 2 above)
Comment (lava) [reply scripts/query-tracing.bcc:14]: Probably, but that's up to the caller and no concern to the script.

File: scripts/query-tracing.bcc (hunk 3 above)
Comment (lava) [reply scripts/query-tracing.bcc:40]: Good point.

File: scripts/query-tracing.bcc (hunk 1 above)
Comment (lava) [reply scripts/query-tracing.bcc:224]: There's no script currently, I did everything so far in interactive mode. I'll just add this as a comment:

```
> data = read.table('queries.txt', sep='\t', header = TRUE)
> data %>% ggplot(aes(x=value, y=variable)) + geom_density_ridges()
```

**Review by tobim — APPROVED** [review]

**Pull request merged**